source/mir/blas/dot.d

```diff
@@ -0,0 +1,127 @@
+/**
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Ilya Yaroshenko
+*/
+module mir.blas.dot;
+
+import mir.ndslice.slice;
+import std.traits;
+
+/++
+Params:
+	x = vector
+	y = vector
+Returns:
+	scalar `xᵀ × y`
+Note:
+	`dot` implementation is naive for now.
++/
+auto dot(V1, V2)(scope V1 x, scope V2 y)
+	if ((isDynamicArray!V1 || is(V1 : Slice!(1, T1*), T1)) &&
+		(isDynamicArray!V2 || is(V2 : Slice!(1, T2*), T2)) &&
+		is(Unqual!(ForeachType!V1) == Unqual!(ForeachType!V2)))
```

Comment:
maybe combining `isDynamicArray!V2 || is(V2 : Slice!(1, T2*), T2)` in a template?
Currently it's not very readable for a user ...